exla/lib/exla/backend.ex

```diff
@@ -0,0 +1,192 @@
+defmodule EXLA.Backend do
+  @moduledoc """
+  EXLA Nx backend implementation for usage of EXLA outside of defn.
+
+  The EXLA backend is designed for rapid prototyping of accelerated Nx programs
+  and ensures consistency from prototype to implementation. Generally, you'll want
+  to experiment with ideas before organizing them into a project. Backends allow
+  you to experiment with Nx in "eager mode" before utilizing the stricter, but
+  more performant constructs such as `defn` and JIT compilation.
+
+  While you can also prototype with the BinaryBackend, the EXLA backend offers
+  the following advantages:
+
+    1) Performance - The EXLA backend executes functions on CPU/GPU/TPU.
+
+    2) Consistency - You may encounter slight inconsistencies in behavior between
+    the BinaryBackend and the EXLA backend which leads to different behavior when
+    changing "eager mode" code to compiled code. Using EXLA in "eager mode" ensures
+    behavior is consistent when transitioning from eager mode to compiled.
+  """
+
+  defstruct [:state]
+
+  @behaviour Nx.Backend
+
+  alias EXLA.Backend, as: B
+  import Nx.Shared
+
+  # For now this behaves like the BinaryBackend; however, we can switch this
+  # behavior to always make tensors as EXLA Buffers, and all of the backend
+  # stuff transfers between them
+
+  @impl true
+  def constant(%{type: type, shape: shape} = out, constant, _backend_options) do
+    data = :binary.copy(number_to_binary(constant, type), Nx.size(shape))
+    from_binary(out, data)
+  end
+
+  defp number_to_binary(number, type),
+    do: match_types([type], do: <<write!(number, 0)>>)
+
+  @impl true
+  def from_binary(t, binary, _backend_options), do: from_binary(t, binary)
+
+  defp from_binary(t, binary) when is_binary(binary), do: %{t | data: %B{state: binary}}
```

Comment:
The state for this should never a binary. It should always be a reference to something on the EXLA device. Please check the previous implementation :)